src/herder/HerderImpl.cpp

```diff
@@ -1321,6 +1360,28 @@ HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger,
     // so this is the most appropriate value to use as closeTime.
     uint64_t nextCloseTime =
         VirtualClock::to_time_t(mApp.getClock().system_now());
+    if (ledgerSeqToTrigger == lcl.header.ledgerSeq + 1)
+    {
+        auto it = mDriftCTSlidingWindow.find(ledgerSeqToTrigger);
+        if (it == mDriftCTSlidingWindow.end())
+        {
+            // Record local close time _before_ it gets adjusted to be valid
+            // below
+            mDriftCTSlidingWindow[ledgerSeqToTrigger] =
+                std::make_pair(nextCloseTime, std::nullopt);
+            while (mDriftCTSlidingWindow.size() > CLOSE_TIME_DRIFT_LEDGER_WINDOW)
+            {
+                mDriftCTSlidingWindow.erase(mDriftCTSlidingWindow.begin());
+            }
+        }
+        else
+        {
+            CLOG_WARNING(Herder,
```

Comment:
Should we assert or throw here instead of warn? Does this indicate a logic bug, or is this possible if we have a longer than expected ledger such that the trigger timer fires multiple times on one ledger?

Reply:
we should handle this as gracefully as possible so the node can continue participating in consensus in case of buggy trigger (then the worst case is delays in nomination, which is still much better than a halt)